### PR TITLE
KAFKA-5839: Upgrade Guide doc changes for KIP-130

### PR DESCRIPTION
**Pull request by fhussonnois**

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -109,6 +109,23 @@
...
+    </ul>
+    <p>
+        Deprecated methods in <code>KafkaStreams</code>:
+    </p>
+    <ul>
+        <li><code>toString()</code></li>
+        <li><code>toString(final String indent)</code></li>
+    </ul>
+    <p>
+        The above methods have been deprecated in favor of using the new method <code>#localThreadsMetadata()</code>.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: We should also point to new class/method `TopologyDescription` and `Topology#describe()` here. Deprecated method `toStrint()` did give static and runtime information and it's functionality got split up in static information via `TopologyDescription` and runtime information provided via `ThreadMetadata`.

**Review by mjsax — COMMENTED** [review]
Thanks for the PR @fhussonnois! Some minor suggestions to improve the docs further.

**Comment by guozhangwang** [comment]
LGTM. I will have minor changes to rearrange the sections while merging the PR.

File: docs/streams/upgrade-guide.html (hunk 1 above)
Comment (guozhangwang) [reply docs/streams/upgrade-guide.html]: Makes sense. @fhussonnois could you update the PR for this comment?

**Comment by guozhangwang** [comment]
Merged to trunk with some minor fixes.

**Pull request closed without merging**